src/middleware/authentication.js

```diff
@@ -0,0 +1,26 @@
+const jwt = require('jsonwebtoken');
+const { AuthenticationError } = require('../error/AuthenticationError');
+
+function authenticateToken(req, res, next) {
+  if (req.headers.cookie == null) {
+    res.send(new AuthenticationError());
+  }
+
+  const authHeader = req.headers.cookie;
+  const token = authHeader && authHeader.split('=')[1];
+
+  if (token == null) {
+    res.send(new AuthenticationError());
+  }
+
+  return jwt.verify(token, process.env.SECRET, (err, player) => {
```

Comment:
middlewares shouldn't return, this cause some issues

Reply:
fixed